boxsdk/client/client.py

```diff
@@ -1300,3 +1300,127 @@ def device_pinners(self, enterprise_id, direction=None, limit=None, marker=None,
             fields=fields,
             return_full_pages=False,
         )
+
+    def metadata_template(self, scope, template_key):
+        """
+        Initialize a :class:`MetadataTemplate` object with the given scope and template key.
+
+        :param scope:
+            The scope of the metadata template, e.g. 'enterprise' or 'global'
+        :type scope:
+            `unicode`
+        :param template_key:
+            The key of the metadata template
+        :type template_key:
+            `unicode`
+        :returns:
+            The metadata template object
+        :rtype:
+            :class:`MetadataTemplate`
+        """
+        return self.translator.get('metadata_template')(
+            session=self._session,
+            object_id='{0}/{1}'.format(scope, template_key),
+        )
+
+    @api_call
+    def get_metadata_template_by_id(self, template_id):
+        """
+        Retrieves a metadata template by ID
+
+        :param template_id:
+            The ID of the template object
+        :type template_id:
+            `unicode`
+        :returns:
+            The metadata template with data populated from the API
+        :rtype:
+            :class:`MetadataTemplate`
+        """
+        url = self._session.get_url('metadata_templates', template_id)
+        response = self._session.get(url).json()
+        return self.translator.get('metadata_template')(
+            session=self._session,
+            object_id='{0}/{1}'.format(response['scope'], response['templateKey']),
+            response_object=response,
+        )
+
+    @api_call
+    def get_metadata_templates(self, scope='enterprise', limit=None, marker=None, fields=None):
+        """
+        Get all metadata templates for a given scope.  By default, retrieves all metadata templates for the current
+        enterprise.
+
+        :param scope:
+            The scope to retrieve templates for
+        :type scope:
+            `unicode`
+        :type limit:
+            `int` or None
+        :param marker:
+            The paging marker to start paging from.
+        :type marker:
+            `unicode` or None
+        :param fields:
+            List of fields to request.
+        :type fields:
+            `Iterable` of `unicode`
+        :returns:
+            The collection of metadata templates for the given scope
+        :rtype:
+            :class:`BoxObjectCollection`
+        """
+        return MarkerBasedObjectCollection(
+            url=self._session.get_url('metadata_templates', scope),
+            session=self._session,
+            limit=limit,
+            marker=marker,
+            fields=fields,
+            return_full_pages=False,
+        )
+
+    @api_call
+    def create_metadata_template(self, display_name, fields, template_key=None, hidden=False, scope='enterprise'):
```

Comment:
I don't mind moving it, but I put it last since at this time only enterprise scope templates can be created.  Nobody calling this method today (and likely for the near future) will be setting that parameter, whereas they will likely want to set the others.